tools/crm_mon.c

```diff
@@ -1272,39 +1273,48 @@ add_output_args(void) {
  */
 static void
 reconcile_output_format(pcmk__common_args_t *args) {
+    // None of the callbacks below can fail, but check for future-proofing
     gboolean retval = TRUE;
-    GError *err = NULL;
 
     if (output_format != mon_output_unset) {
         return;
     }
 
+    /* @COMPAT: Use case-sensitive comparisons, and use default format only if
+     * pcmk__str_eq(args->output_ty, <default>, pcmk__str_null_matches).
+     * Let pcmk__output_new() throw a pcmk_rc_unknown_format if an invalid
+     * format is specified with --output-as.
+     */
```

Comment:
That's precisely the point -- currently, if we find a case-insensitive match (e.g., `"XML"`), we overwrite it with the lowercase match (e.g., `"xml"`) in this function before we get to `pcmk__output_new()`. So it doesn't fail.

Further, currently if we don't match "(html|xml|text)" case-insensitively, then we default to "console" (or "text" if no curses). In any case, we get a valid format name, but it may not be what we asked for:
* "XML" gets us "xml"
* "asdf" gets us "console"

If we implemented the case-sensitive matching, we'd have:
* "XML" gets us "console"

If we implemented the whole COMPAT note, we'd have:
* "XML" gets us an error
* "asdf" gets us an error

Reply:
I think we can go ahead. The overwriting code was about remapping deprecated arguments, not capitalized values with the new arguments, so it's more a bug. Also, --output-as is still relatively new so I wouldn't expect a whole lot of opportunity for misuse yet. I'd rather crm_mon --output-as not behave differently than any other command.

Reply:
> The overwriting code was about remapping deprecated arguments

The opposite: `reconcile_output_format()` returns immediately if any of the deprecated arguments were used. It's about mimicking the behavior of the deprecated arguments with the new ones.

> I think we can go ahead.

Works for me. In that case, my preference is to do everything explicitly in `reconcile_output_format()` and not call the callbacks. The callbacks don't do anything useful for us here as you'll see, and they won't change since they're used only with deprecated arguments.

Reply:
Deprecated arguments currently (in main and in this PR) take precedence over `--output-as` arguments. We bail out of `reconcile_output_formats()` if a deprecated argument was parsed. Let me know if that should change.

Reply:
> The opposite: reconcile_output_format() returns immediately if any of the deprecated arguments were used. It's about mimicking the behavior of the deprecated arguments with the new ones.

Oh right. In any case, that code originated in the `safe_str_eq()` days, so case-insensitive wasn't an intentional choice, it just carried over from that. I'd still consider it a bug.

> Deprecated arguments currently (in main and in this PR) take precedence over --output-as arguments. 

IIRC there may have been an issue in detecting when --output-as was explicitly specified vs defaulting. Not worth any effort.